src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java

```diff
@@ -1327,6 +1330,41 @@ public ClassSymbol(long flags, Name name, Symbol owner) {
             this.type.tsym = this;
         }
 
+        public void addPermittedSubclass(ClassSymbol csym, int pos) {
+            if (isPermittedExplicit) {
```

Comment:
If I understand it properly, the codepaths to add implicitly and explicitly declared permitted subclasses are completely different. And when the permitted subclasses are explicit, the `addPermittedSubclass` won't be called.

For consideration: just using `Assert.check(!isPermittedExplicit)`, drop the if.

Reply:
sure will do, thanks